### PR TITLE
doc(quickstart.md,configuring-object-storage.md): start documenting object storage configs

### LINKED ISSUE #3: [storage] Design for Off-cluster Storage
For the alpha, we need to point a deis installation at off-cluster storage. This ticket is a placeholder for that design.

* [x] design off-cluster storage configuration (https://github.com/deis/builder/pull/21)
* [ ] document storage configuration in deis/workflow (see #63) _note: https://github.com/deis/workflow/pull/108 has the fix for this, and it was moved to the `v2.0-beta1` milestone._

### LINKED ISSUE #63: document how to configure storage
docs should include

- how workflow & builder detect minio
- the env vars to set in order to use external object storage (#3)

### PR DESCRIPTION
**Pull request by arschles**
Fixes #63 
Fixes #3

(note that in #3, this closes the second and final bullet point)

### CODE REVIEW

**Comment by arschles** [comment]
moved to `v2.0-beta1` and will comment on the corresponding issues

cc @technosophos

**Comment by jchauncey** [comment]
:+1:

File: docs/src/installing-deis/configuring-object-storage.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+- [slugrunner](https://github.com/deis/slugrunner)
+- [registry](https://github.com/deis/registry)
+
+These components are built flexibly, so they can work out of the box with almost any system that is compatible with the [S3 API](http://docs.aws.amazon.com/AmazonS3/latest/API/APIRest.html).
+
+# Minio
+
+Additionally, Deis ships with a [Minio](http://minio.io) [component](https://github.com/deis/minio). This component runs as a Kubernetes service, and the components listed above are configured to automatically look for that service and use it as object storage if it's available.
+
+# Telling Deis What to Use
```
Comment (carmstrong) [inline docs/src/installing-deis/configuring-object-storage.md]: Should we omit this section entirely until we have content, or are we okay with this TODO?

File: docs/src/installing-deis/configuring-object-storage.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+The Deis components determine what object storage system to use via environment variables that you set up:
+
+TODO: list the lookup order
+
+# Limitations
+
+- In some cases, the minio service must be started up before other components, so that they see the Minio service's environment variables. If you use [helm](https://github.com/helm/helm) to install deis, this ordering won't be ensured. (builder, slugbuilder and slugrunner are fine)
```
Comment (technosophos) [inline docs/src/installing-deis/configuring-object-storage.md]: What do you mean "won't be ensured" here?

File: docs/src/installing-deis/configuring-object-storage.md (hunk 1 above)
Comment (arschles) [inline docs/src/installing-deis/configuring-object-storage.md]: forgot about this TODO, I'll fill it in now that I have the lookup order. thanks @carmstrong

File: docs/src/installing-deis/configuring-object-storage.md (hunk 2 above)
Comment (arschles) [inline docs/src/installing-deis/configuring-object-storage.md]: I mean that a component that depends on minio may be installed before the minio service is up. Currently, registry is the only component that would suffer in this scenario.

**Comment by arschles** [comment]
@carmstrong listed the lookup order

**Comment by arschles** [comment]
@technosophos are you ok with the "this ordering won't be ensured" language?

File: docs/src/installing-deis/configuring-object-storage.md (hunk 2 above)
Comment (technosophos) [inline docs/src/installing-deis/configuring-object-storage.md]: I don't see how that would happen with Helm, given that services are installed before RCs, daemonsets, and pods.

**Comment by technosophos** [comment]
@arschles I don't understand the issue you are raising. We've never seen that happen, as services are always installed prior to anything that would use them. It could happen if you were _not_ using Helm... but with Helm it should not.

**Comment by arschles** [comment]
nevermind, I forgot about the ordering point that you raised in https://github.com/deis/workflow/pull/108#discussion_r48189231.

File: docs/src/installing-deis/configuring-object-storage.md (hunk 2 above)
Comment (arschles) [inline docs/src/installing-deis/configuring-object-storage.md]: ah, I forgot about the ordering. I'll remove that language

**Pull request merged**